### PR TITLE
MetricsCounter

### PR DESCRIPTION
**Pull request by bobokvsky**

### CODE REVIEW

File: lib/two_stage_pipeliner/metrics_counters/classification.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+class ClassificationMetricsCounter(MetricsCounter):
+    def __init__(self, inferencer: ClassificationInferencer):
+        assert isinstance(inferencer, ClassificationInferencer)
+        MetricsCounter.__init__(self, inferencer)
+
+    def score(self, data_generator: BatchGeneratorBboxData) -> pd.DataFrame:
+        n_true_bboxes_data = data_generator.data
+        n_pred_bboxes_data = self.inferencer.predict(data_generator)
+        true_bboxes_data = [bbox_data for bboxes_data in n_true_bboxes_data for bbox_data in bboxes_data]
```
Comment (SlinkoIgor) [inline lib/two_stage_pipeliner/metrics_counters/classification.py:17]: какая-то очень сложная конструкция. я бы разбил. не могу проверить, правильно ли тут все распаковывается

File: lib/two_stage_pipeliner/metrics_counters/core/classification.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+import pandas as pd
+from sklearn.metrics import classification_report
+
+
+def get_df_classifier_metrics(true_labels: List[str],
+                              pred_labels: List[str]) -> pd.DataFrame:
+    df_classifier_metrics = pd.DataFrame(
+        classification_report(true_labels, pred_labels, output_dict=True)
+    ).drop(columns='accuracy').T
```
Comment (SlinkoIgor) [inline lib/two_stage_pipeliner/metrics_counters/core/classification.py:11]: а мне нравится колонка accuracy )

File: lib/two_stage_pipeliner/metrics_counters/core/detection.py [hunk 3]
```diff
@@ -0,0 +1,399 @@
...
+        'iou_mean': [round(iou_mean, 3)],
+        **coco_metrics
+    }).T
+    df_detector_metrics.columns = ['value']
+    df_detector_metrics = df_detector_metrics.astype(object)
+
+    return df_detector_metrics
+
+
+def get_df_detector_metrics_recall(
```
Comment (SlinkoIgor) [inline lib/two_stage_pipeliner/metrics_counters/core/detection.py:309]: по названию не очень понятно, чем отличается от предыдущей функции. мб надо описание написать...
по ощущением, это какое-то подмножество предыдущей функции. 
кажется тут понадобится серьезный рефакторинг в ближайшем будущем: кода много. Но мне сложно понять, что он делает

File: lib/two_stage_pipeliner/metrics_counters/detection.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        n_true_images_data = data_generator.data
+
+        for image_data in n_true_images_data:
+            assert image_data.bboxes_data is not None
+            for bbox_data in image_data.bboxes_data:
+                assert bbox_data.xmin is not None
+                assert bbox_data.ymin is not None
+                assert bbox_data.xmax is not None
+                assert bbox_data.ymax is not None
+                assert bbox_data.label is not None
```
Comment (SlinkoIgor) [inline lib/two_stage_pipeliner/metrics_counters/detection.py:30]: мб сделаем функцию is_valid() в классе bbox_data ?
